editor.planx.uk/src/pages/FlowEditor/lib/store/user.ts

```diff
@@ -23,69 +17,30 @@ export const userStore: StateCreator<UserStore, [], [], UserStore> = (
   set,
   get,
 ) => ({
-  id: 0,
-  firstName: "",
-  lastName: "",
-  email: "",
-  isPlatformAdmin: false,
-  teams: [],
 
-  setUser: (user: User) =>
-    set({
-      id: user.id,
-      firstName: user.firstName,
-      lastName: user.lastName,
-      email: user.email,
-      isPlatformAdmin: user.isPlatformAdmin,
-      teams: user.teams,
-    }),
+  setUser: (user: User) => set({ user }),
 
-  getUser: () => ({
-    id: get().id,
-    firstName: get().firstName,
-    lastName: get().lastName,
-    email: get().email,
-    isPlatformAdmin: get().isPlatformAdmin,
-    teams: get().teams,
-  }),
+  getUser: () => get().user,
 
-  canUserEditTeam: (teamSlug) => {
+  canUserEditTeam(teamSlug) {
+    const user = this.getUser();
+    if (!user) return false;
+    
     return (
-      get().isPlatformAdmin ||
+      user.isPlatformAdmin ||
       teamSlug === "templates" ||
-      get().teams.filter(
+        user.teams.filter(
         (team) => team.role === "teamEditor" && team.team.slug === teamSlug,
       ).length > 0
     );
   },
 
   async initUserStore(jwt: string) {
-    const email = (jwtDecode(jwt) as any)["email"];
-    const users = await client.query({
-      query: gql`
-        query GetUserByEmail($email: String!) {
-          users: users(where: { email: { _eq: $email } }) {
-            id
-            firstName: first_name
-            lastName: last_name
-            email
-            isPlatformAdmin: is_platform_admin
-            teams {
-              role
-              team {
-                name
-                slug
-                id
-              }
-            }
-          }
-        }
-      `,
-      variables: { email },
-    });
+    if (this.getUser()) return;
```

Comment:
Checking this means we now don't run this query on each route change, just on initial page load.